.github/workflows/coverage.yml

```
@@ -0,0 +1,40 @@
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+
+      - name: Set up Gradle
+        uses: gradle/gradle-build-action@v2
+
+      - name: Run tests and generate coverage report
+        run: ./gradlew koverXmlReport
+
+      - name: Extract coverage from XML
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:3:32: Double quote to prevent globbing and word splitting

(shellcheck)